RTCError and RTCErrorEvent docs (#44047)
mdn/content: files/en-us/web/api/rtcerrorevent/error/index.md

{{APIRef("WebRTC")}}

The **`error`** read-only property of the {{domxref("RTCErrorEvent")}} interface contains an {{domxref("RTCError")}} object that describes the {{Glossary("WebRTC")}}-specific details of the error.

## Value

> An {{domxref("RTCError")}} object whose properties provide RTC-specific details about the error.

Reviewer: ```suggestion
An {{domxref("RTCError")}} object.
```

## Examples

### Basic usage

In this example, a handler is established for an {{domxref("RTCDataChannel")}}'s {{domxref("RTCDataChannel.error_event", "error")}} event.

```js
dataChannel.addEventListener("error", (event) => {
  let error = event.error;

  if (error.errorDetail === "sdp-syntax-error") {
    const errLine = error.sdpLineNumber;
    const errMessage = error.message;

    const alertMessage = `A syntax error occurred interpreting line ${errLine} of the SDP: ${errMessage}`;
    showMyAlertMessage("Data Channel Error", alertMessage);
  } else {
    terminateMyConnection();
  }
});
```

> If the error is an SDP syntax error—indicated by its {{domxref("RTCError.errorDetail", "errorDetail")}} property being `sdp-syntax-error`— a message string is constructed to present the error message and the line number within the SDP at which the error occurred.

Reviewer: ```suggestion
If the error is an SDP syntax error—indicated by its {{domxref("RTCError.errorDetail", "errorDetail")}} property being `sdp-syntax-error`— a message string is constructed to present the error message and the SDP message line number where the error occurred.
```
